python/lsst/analysis/tools/actions/keyedData/calcDistances.py

```diff
@@ -140,12 +148,29 @@ def _compressArray(arrayIn):
         width = (self.width * u.arcmin).to_value(u.degree)
         annulus = D + (width / 2) * np.array([-1, +1])
 
-        # Match this catalog to itself within the radius and then cut
-        # to the annulus inner radius.
-        with Matcher(meanRa, meanDec) as m:
-            idx, i1, i2, d = m.query_self(annulus[1], return_indices=True)
+        if annulus[1] >= self.bruteForceRadiusThreshold:
+            # Use brute-force matching.
+            if len(meanRa) > self.bruteForceMaxMatch:
+                bruteForceSelection = rng.choice(len(meanRa), size=self.bruteForceMaxMatch, replace=False)
+            else:
+                bruteForceSelection = np.arange(len(meanRa))
+
+            i1 = np.tile(bruteForceSelection, len(bruteForceSelection))
+            i2 = np.repeat(bruteForceSelection, len(bruteForceSelection))
+
+            d = np.rad2deg(
+                sphDist(
+                    np.deg2rad(meanRa)[i1],
+                    np.deg2rad(meanDec)[i1],
+                    np.deg2rad(meanRa)[i2],
+                    np.deg2rad(meanDec)[i2],
+                ),
+            )
+        else:
+            with Matcher(meanRa, meanDec) as m:
+                idx, i1, i2, d = m.query_self(annulus[1], return_indices=True)
```

Comment:
I see what you're doing here and it is ok, but I wonder why you use a brute force search after randomly selecting objects to match. Since you've reduced the number of observations can't you just use smatch.Matcher now? I would think that it would have gotten rid of the memory issues.

Reply:
At 3 degrees it turns out that the brute force uses less memory and time than actually trying to construct the tree and create all the cross-indexes.  So the Matcher would be fine, but I think this is better and has guaranteed predictability.

Reply:
I considered also putting in a check if the radius was larger than the size of the box but deemed that over-optimization.